src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -41,6 +41,8 @@
     # JSON Operators: https://www.postgresql.org/docs/9.5/functions-json.html
     [
         StringLexer("right_arrow", "=>", CodeSegment),
+        StringLexer("question_mark", "?", CodeSegment),
+        RegexLexer("atsign_literal", r"@[a-zA-Z_][\w]*", CodeSegment),
```

Comment:
I moved this from this:
```py
RegexLexer("atsign_literal", r"[@][\w]*", CodeSegment),
```

For the following reasons:
- It's not just the `atsign` but `atsign` followed by literal identifier. So this is a better name to make that obvious
- In BigQuery literals should start with a character or an underscore so `@1abc` is not valid but would be accepted by the original regex.
- It prevents just `@` being accepted.

I've committed that change and added test cases so this is just an FI an no further action required.